experimental/go-modules/image/project/Dockerfile

```diff
@@ -0,0 +1,15 @@
+# Install the app dependencies in a full Node docker image
+FROM golang:alpine
+
+
+# Install user-app dependencies
+WORKDIR "/project/user-app"
+RUN apk add --update git bash
+RUN go get -u github.com/golang/dep/cmd/dep
+COPY ./user-app/* ./
```

Comment:
some go projects speed up the build by cacheing the dep layer seperately from the app:
```
COPY go.* ./
RUN go mod download
COPY . .
RUN go build // etc.
```